src/session/session_helper.c

```diff
@@ -134,3 +134,35 @@ __wt_session_dump(WT_SESSION_IMPL *session, WT_SESSION_IMPL *dump_session, bool
     __wt_scr_free(session, &buf);
     return (ret);
 }
+
+/*
+ * __wt_session_set_last_error --
+ *     Stores information about the last error to occur during this session.
+ */
+int
+__wt_session_set_last_error(
+  WT_SESSION *wt_session, int err, int sub_level_err, const char *err_msg_content)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+
+    char *err_msg = session->err_info.err_msg;
+    size_t err_msg_size;
+
+    /* Ensure arguments are valid, and that session is not a mock */
+    WT_ASSERT(session, __wt_is_valid_sub_level_error(sub_level_err));
```

Comment:
We should assert the session->err_info is non-null before dereferencing the struct.

Reply:
How should I handle mock sessions? This function will get called for them (unless I add some extra logic in err.c), and they don't have err_info.

Reply:
session->err_info is a struct, not a pointer to struct, so at worst it'll just be empty